onnxruntime/core/graph/graph.cc

```diff
@@ -1629,7 +1629,7 @@ Status Graph::BuildConnections(std::unordered_set<std::string>& outer_scope_node
             // if it is present there and not explicitly listed in the ordered graph outputs
             // (as that implies we should leave it as an output).
             // If the Graph was loaded from a GraphProto, honor the explicit graph outputs and leave as is.
-            if (!is_loaded_from_model_file_) {
+            if (!is_loaded_from_model_file_ && !graph_outputs_manually_set_) {
```

Comment:
made this change to allow manually setting a graph output that is also consumed by a subgraph